rm-datasource/src/main/java/io/seata/rm/datasource/ConnectionProxy.java

```diff
@@ -185,6 +185,9 @@ public void commit() throws SQLException {
                 return null;
             });
         } catch (SQLException e) {
+            if (targetConnection != null && !getAutoCommit()) {
+                rollback();
```

Comment:
```
if(autocommit=false) {
    rollback();
}
```
What kind of problem of this line solved ?

Reply:
this connection is reset on rollback to prevent duplicate registration and connection unavailability issues

Reply:
How about under this situation 
```
set autocommit = 0;
start transaction;
//success
insert into account values (xxxx);
savepoint step1;
//lock conflict
insert into order values(xxxxx);
rollback to step1;
```